ui/src/app/pages/admin/disks-table.spec.tsx

```diff
@@ -0,0 +1,200 @@
+  mockCromwellDisk,
+  mockRStudioDisk,
+];
+
+let mockdisksAdminApi;
+
+beforeEach(() => {
+  jest.mock('app/services/swagger-fetch-clients');
+  mockdisksAdminApi = jest.spyOn(swaggerClients, 'disksAdminApi');
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+const setup = (mockOverrides) => {
+  mockdisksAdminApi.mockImplementation(() => ({
+    ...mockOverrides,
+  }));
+};
+
+test('loads and displays table', async () => {
+  setup({ listDisksInWorkspace: () => Promise.resolve(mockDisks) });
+  render(<DisksTable sourceWorkspaceNamespace='123' />);
+  await waitForElementToBeRemoved(() =>
+    screen.getByTitle('disks loading spinner')
+  );
+
+  mockDisks.forEach((disk) => {
+    const row = screen.getByText(disk.name).closest('tr');
+    const rowScope = within(row);
+    expect(rowScope.getByText(disk.name)).toBeInTheDocument();
+    expect(rowScope.getByText(disk.creator)).toBeInTheDocument();
+    expect(
+      rowScope.getByText(convertDate(disk.createdDate))
+    ).toBeInTheDocument();
+    expect(rowScope.getByText(disk.status)).toBeInTheDocument();
+    expect(
+      rowScope.getByText(getEnvironmentType(disk.isGceRuntime, disk.appType))
+    ).toBeInTheDocument();
+    expect(rowScope.getByText(disk.size)).toBeInTheDocument();
+    expect(
+      rowScope.getByText('Delete').closest('div[role="button"]')
+    ).not.toHaveStyle(`cursor: not-allowed`);
+  });
+});
+
+test('loads and displays empty table', async () => {
+  setup({ listDisksInWorkspace: () => Promise.resolve([]) });
+  render(<DisksTable sourceWorkspaceNamespace='123' />);
+  await waitForElementToBeRemoved(() =>
+    screen.getByTitle('disks loading spinner')
+  );
+  expect(screen.getByText('No disks found')).toBeInTheDocument();
+});
+
+test('show spinner while disks are loading', async () => {
+  setup({
+    listDisksInWorkspace: () => new Promise(() => {}),
+  });
+  render(<DisksTable sourceWorkspaceNamespace='123' />);
+
+  expect(screen.getByTitle('disks loading spinner')).toBeInTheDocument();
+});
+
+test('delete disk', async () => {
+  const mockDeleteFunction = jest.fn(() => Promise.resolve({}));
+  const mockListFunction = jest.fn(() => Promise.resolve(mockDisks));
+
+  setup({
+    deleteDisk: mockDeleteFunction,
+    listDisksInWorkspace: mockListFunction,
+  });
+
+  render(<DisksTable sourceWorkspaceNamespace='123' />);
+  await waitForElementToBeRemoved(() =>
+    screen.getByTitle('disks loading spinner')
+  );
+
+  const row = screen.getByText(mockJupyterDisk.name).closest('tr');
+  const rowScope = within(row);
+  const jupyterDeleteButton = rowScope
+    .getByText('Delete')
+    .closest('div[role="button"]');
+
+  expect(mockListFunction).toHaveBeenCalledTimes(1);
+  fireEvent.click(jupyterDeleteButton);
+  await waitFor(() => {
+    expect(mockDeleteFunction).toHaveBeenCalledTimes(1);
+    expect(mockListFunction).toHaveBeenCalledTimes(2);
+  });
+});
+
+test('disable delete buttons while deleting a disk', async () => {
```

Comment:
I'm glad that you brought this up! I ended up finding a solution that I really like and I think it will be useful in the future. Take a look for resolveDeleteFunction in ui/src/app/pages/admin/disks-table.spec.tsx. 

I referenced this to set it up:
https://stackoverflow.com/questions/26150232/resolve-javascript-promise-outside-the-promise-constructor-scope